bump tendermint 0.35

altergui opened the pull request:
this was the original upgrade in january v0.34.13 -> v0.35.1
 * https://github.com/vocdoni/vocdoni-node/pull/429
 * https://github.com/vocdoni/vocdoni-node/pull/455

a week later, @p4u rolled back to tendermint v0.34.13
 * https://github.com/vocdoni/vocdoni-node/pull/468
 
 now i'm reverting that last rollback, i.e. upgrading again to v0.35.1

altergui committed:
update tendermint to  0.35

altergui committed:
vochaininspector: no-op for listBlockVotes, store module missing in tendermint v0.35

altergui committed:
go mod tidy

altergui committed:
revert code mess introduced in 1a05e449: 'vochain: update genesis and make staticcheck happy'

altergui committed:
voteID was only used in listBlockVotes

altergui committed:
bump tendermint v0.35.1 -> v0.35.9

altergui committed:
fixup! update tendermint to  0.35

app.chainId -> app.chainID

mvdan commented on cmd/vochaininspector/inspector.go:
```diff
@@ -193,64 +187,65 @@ func newVochain(chain, dataDir string) *vochain.BaseApplication {
+	// 	vid, err := voteID(vote.ProcessId, vote.Nullifier)
+	// 	if err != nil {
+	// 		log.Fatalf("cannot get voteID: %v", err)
+	// 	}
```
delete the code instead? leaving it commented out doesn't help - it's always in the git history if we need it.

altergui replied:
good catch: actually a no-op cli flag was being left behind, cleaned everything up in 1fadf6c

mvdan commented on vochain/start.go:
```diff
@@ -376,251 +319,34 @@ func newTendermint(app *BaseApplication,
-			log.Warnf("failed to get pubkey from private validator: %v", err)
-		}
-		log.Infof("tendermint address: %v", pubKey.Address())
+	/*	if len(localConfig.PrivValidatorListenAddr) > 0 {
```
same

altergui committed:
dvotenode: remove support for hardware wallets (flag vochainPrivValidator)

this feature was unused, so we didn't care enough to port the code
(didn't even check if tendermint v0.35 actually supports this anymore)

altergui committed:
vochaininspector: drop listBlockVotes command, needs rewrite to work on tendermint v0.35
